src/builder-module.c

```diff
@@ -1502,12 +1523,220 @@ find_file_with_extension (GFile *dir,
+          g_autoptr(GFile) license_file = NULL;
+
+          if (!g_str_has_prefix (name, pattern))
+            continue;
+
+          license_file = g_file_get_child (source_dir, name);
+          g_ptr_array_add (files, g_steal_pointer (&license_file));
+          break;
+        }
+    }
+
+  if (my_error != NULL)
+    {
+      g_propagate_error (error, g_steal_pointer (&my_error));
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static gboolean
+find_license_files (BuilderModule  *self,
+                    GFile          *source_dir,
+                    GPtrArray     **license_files_out,
+                    GError        **error)
+{
+  g_autoptr(GPtrArray) license_files =
+    g_ptr_array_new_with_free_func (g_object_unref);
+
+  if (self->license_files)
+    {
+      if (!find_defined_license_files (self->license_files,
+                                       source_dir,
+                                       license_files,
+                                       error))
+        return FALSE;
+    }
+  else
+    {
+      if (!find_default_license_files (self,
+                                       source_dir,
+                                       license_files,
+                                       error))
+        return FALSE;
+    }
+
+  if (license_files_out)
+    *license_files_out = g_steal_pointer (&license_files);
+  return TRUE;
+}
+
+static GFile *
+get_license_dst_file (GFile *license_dir,
+                      GFile *license_file)
+{
+  g_autofree char *license_file_name = g_file_get_basename (license_file);
+  const char *candidate = license_file_name;
+  int i = 1;
+
+  while (TRUE)
+    {
+      g_autoptr(GFile) dst = NULL;
+      g_autofree char *owned_candidate = NULL;
+
+      if (!candidate)
+        {
+          owned_candidate = g_strdup_printf ("LICENSE_%u", i + 1);
+          candidate = owned_candidate;
+          i++;
+        }
+
+      dst = g_file_get_child (license_dir, candidate);
+      if (!g_file_query_exists (dst, NULL))
+        return g_steal_pointer (&dst);
+
+      candidate = NULL;
+    }
+}
+
+static gboolean
+builder_module_install_licenses (BuilderModule  *self,
+                                 const char     *id,
+                                 GFile          *source_dir,
+                                 GFile          *app_dir,
+                                 GError        **error)
+{
+  g_autoptr(GPtrArray) license_files = NULL;
+  g_autoptr(GFile) license_dir = NULL;
+
+  license_dir = g_file_new_build_filename (g_file_get_path (app_dir),
+                                           "files/share/licenses",
+                                           id,
+                                           self->name,
+                                           NULL);
+  if (!flatpak_mkdir_p (license_dir, NULL, error))
+    return FALSE;
+
+  if (!find_license_files (self, source_dir, &license_files, error))
+    return FALSE;
```

Comment:
i'd reorder this a bit, otherwise it is creating a bunch of empty directories with nothing in them

Reply:
i think it happened to me when the module to build is already cached so source dir ends up empty except the post install stage, but I think it makes sense to verify that it had something to install before creating the directory

Reply:
Sure, changed it.